ENT-13540: time_based_vars.cf: refactored acceptance test

larsewi opened the pull request:
Refactored test to print debug information when test fails. This revealed why the test fails every now and then.

It assumed that it would contain the 24th hour instead of the 0th. Now both are accepted. It also appears the regex for `hr` and `hr_2` was mixed up.

```
R: FAIL: "default:sys.time_based_gmt_hr => GMT_Hr00" did not match "^GMT_Hr([1-9]|1[0-9]|2[0-4])$"
R: FAIL: "default:sys.time_based_gmt_hr_2 => GMT_Hr0" did not match "^GMT_Hr(0[1-9]|1[0-9]|2[0-4])$"
R: FAIL: "default:sys.time_based_gmt_hr_qoh => GMT_Hr00_Q1" did not match "^GMT_Hr(0[1-9]|1[0-9]|2[0-4])_Q[1-4]$"
```

Ticket: ENT-13540
Signed-off-by: Lars Erik Wik <lars.erik.wik@northern.tech>

[![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=13180)](https://ci.cfengine.com/job/pr-pipeline/13180/)

larsewi committed:
time_based_vars.cf: refactored acceptance test

Refactored test to print debug information when test fails. This will
hopefully reveal why the test fails every now and then.

Ticket: ENT-13540
Signed-off-by: Lars Erik Wik <lars.erik.wik@northern.tech>

craigcomstock reviewed (approved):
so lovely. thank you. :)

larsewi committed:
time_based_vars.cf: fixed test assumptions

It assumed that it would contain the 24th hour instead of the 0th. Now
both are accepted. It also appears the regex for `hr` and `hr_2` was
mixed up.

```
R: FAIL: "default:sys.time_based_gmt_hr => GMT_Hr00" did not match "^GMT_Hr([1-9]|1[0-9]|2[0-4])$"
R: FAIL: "default:sys.time_based_gmt_hr_2 => GMT_Hr0" did not match "^GMT_Hr(0[1-9]|1[0-9]|2[0-4])$"
R: FAIL: "default:sys.time_based_gmt_hr_qoh => GMT_Hr00_Q1" did not match "^GMT_Hr(0[1-9]|1[0-9]|2[0-4])_Q[1-4]$"
```

Ticket: ENT-13540
Signed-off-by: Lars Erik Wik <lars.erik.wik@northern.tech>

larsewi commented:
[![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=13185)](https://ci.cfengine.com/job/pr-pipeline/13185/)

larsewi commented:
[![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=13198)](https://ci.cfengine.com/job/pr-pipeline/13198/)
